content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1433 @@
+  ipcRenderer.on('update-domain-verification-data', callback)
+}
+```
+
+Finally, the code decodes the account's `Domain` field, if present, and performs domain verification using the method imported above.
+
+4. Update the view logic - in `view/template.html` add the following lines just before the `<input>` element with `id="input-destination-address`:
+
+```html
+<div class="input-group mb-3">
+    <div class="accountVerificationIndicator">
+        <span>Verification status:</span>
+    </div>
+    <input type="text" class="form-control" placeholder="rn95xwUymaMyzAKnZUGuynjZ6qk9RzV4Q7" id="input-destination-address">
+    <!-- Step 8 code additions - start -->
+    <span class="input-group-text">To (Address)</span>
+    <!-- Step 8 code additions - start -->
+</div>
+```
+
+5. Lastly, modify the renderer as described below:
+
+```javascript
+modalButton.addEventListener('click', () => {
+    modalDialog.show()
+})
+
+// Step 8 code additions - start
+const accountVerificationEl = document.querySelector('.accountVerificationIndicator span')
+// Step 8 code additions - end
+
+const destinationAddressEl = document.getElementById('input-destination-address')
+const destinationTagEl = document.getElementById('input-destination-tag')
+const amountEl = document.getElementById('input-xrp-amount')
+const sendXrpButtonEl = document.getElementById('send-xrp-submit-button')
+
+// Step 8 code additions - start
+destinationAddressEl.addEventListener('input', (event) => {
+    window.electronAPI.onDestinationAccountChange(destinationAddressEl.value)
+})
+
+window.electronAPI.onUpdateDomainVerificationData((_event, result) => {
+    accountVerificationEl.textContent = `Domain: ${result.domain || 'n/a'} Verified: ${result.verified}`
+})
+// Step 8 code additions - end
+
+sendXrpButtonEl.addEventListener('click', () => {
+    modalDialog.hide()
+    const destinationAddress = destinationAddressEl.value
+    const destinationTag = destinationTagEl.value
+    const amount = amountEl.value
+
+    window.electronAPI.onClickSendXrp({destinationAddress, destinationTag, amount})
+})
+```
+
+
+The updated preloader `view/8_preloader.js` is also modified the same way by adding the following two event listeners:
+
+`view/8_preload.js`
+{{ include_code("_code-samples/build-a-wallet/desktop-js/view/8_preload.js", language="js", lines="33-38") }}
+
+Run the following command:
+
+```console
+npm run start
+```
+
+Test your wallet app the same way you did in the previous steps. It should display a hint about the receiving account when opening up the "Send XRP" dialog and entering the address. To test domain verification, try entering the following addresses in the "To" box of the Send XRP dialog:
+
+| Address                              | Domain       | Verified? |
+|:-------------------------------------|:-------------|:----------|
+| `rsA2LpzuawewSBQXkiju3YQTMzW13pAAdW` | `mduo13.com` | ✅ Yes    |
+| `rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn` | `xrpl.org`   | ❌ No     |
```

Comment:
For this example, it says `domain domain` instead of `domain: xrpl.org` - 
<img width="576" alt="image" src="https://github.com/XRPLF/xrpl-dev-portal/assets/19694186/92528060-d031-4d37-9984-9ddcbf2389ca">

Reply:
This is the payload from the request: 

```json
{
  Account: 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn',
  Balance: '2939037790',
  Domain: '7872706C2E6F7267',
  Flags: 0,
  LedgerEntryType: 'AccountRoot',
  OwnerCount: 0,
  PreviousTxnID: 'A60DA1A18654422FB7FAC14F405AEFAE9B90B3C773978EDC702FA6AA1D37C292',
  PreviousTxnLgrSeq: 38126630,
  Sequence: 2,
  index: '13F1A95D7AAB7108D5CE7EEAF504B2894B8C674E6D68499076441C4837282BF8'
}
```

Not sure if this is an error on our end...

Reply:
I think it's an error on our end because the domain '7872706C2E6F7267' decodes to `xrpl.org` via ascii, so I think that's fine. It showing up as `domain` is odd.

Reply:
Ah, here it is:

![image](https://github.com/XRPLF/xrpl-dev-portal/assets/102560752/f3121412-ec75-48b2-b0db-69be6064b4c5)

Reply:
I leave it open, should xrpl.org return a .toml file? I added a status check, so the displayed result is correct now. I assumed xrpl.org would return a .toml file, is that correct?

Reply:
Looks good now :)